packages/js-sdk/src/envd/api.ts

```diff
@@ -98,14 +98,16 @@ class EnvdApiClient {
   readonly version: string | undefined
 
   constructor(
-    config: Pick<ConnectionConfig, 'apiUrl' | 'logger'>,
+    config: Pick<ConnectionConfig, 'apiUrl' | 'logger' | 'accessToken'> & { fetch?: (request: Request) => ReturnType<typeof fetch>, headers?: Record<string, string> },
     metadata: {
       version?: string
-    }
+    },
   ) {
     this.api = createClient({
       baseUrl: config.apiUrl,
-      // keepalive: true, // TODO: Return keepalive
+      fetch: config?.fetch,
+      headers: config?.headers,
+      keepalive: true,
```

Comment:
this works fine now, right?

I would do separate PR anyway, so there's better observability